package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,358 @@
+        """Read next auxiliary step and update ``auxstep``.
+
+        Returns
+        -------
+        AuxStep object
+            Updated with the data for the new step.
+
+        Raises
+        ------
+        StopIteration
+            When end of auxiliary data set is reached.
+        """
+        auxstep = self.auxstep
+        new_step = self.step + 1
+        if new_step < self.n_steps:
+            auxstep._data = {term: self.auxdata[term][self.step + 1]
+                             for term in self.terms}
+            auxstep.step = new_step
+            return auxstep
+        else:
+            self.rewind()
+            raise StopIteration
+
+    def _go_to_step(self, i: int) -> EDRStep:
+        """ Move to and read i-th auxiliary step.
+
+        Parameters
+        ----------
+        i: int
+            Step number (0-indexed) to move to
+
+        Returns
+        -------
+        :class:`EDRStep`
+
+        Raises
+        ------
+        ValueError
+            If step index not in valid range.
+        """
+        if i >= self.n_steps or i < 0:
+            raise ValueError("Step index {0} is not valid for auxiliary "
+                             "(num. steps {1})".format(i, self.n_steps))
+        self.auxstep.step = i - 1
+        self.next()
+        return self.auxstep
+
+    def read_all_times(self) -> np.ndarray:
+        """ Get list of time at each step.
+
+        Returns
+        -------
+        NumPy array of float
+            Time at each step.
+        """
+        return self.auxdata[self.time_selector]
+
+    def return_data(self, data_selector: Union[str, List[str], None] = None) \
+            -> Dict[str, np.ndarray]:
+        """ Returns the auxiliary data contained in the :class:`EDRReader`.
+        Returns either all data or data specified as `data_selector` in form
+        of a str or a list of any of :attr:`EDRReader.terms`. `Time` is
+        always returned to allow easy plotting. """
+        if not data_selector or data_selector == "*":
+            return self.auxdata
+        elif isinstance(data_selector, list):
+            data_dict = {"Time": self.auxdata["Time"]}
+            for term in data_selector:
+                if term not in self.terms:
+                    raise KeyError(f"data_selector {term} invalid. Check the "
+                                   "EDRReader's `terms` attribute.")
+                data_dict[term] = self.auxdata[term]
+            return data_dict
+        elif isinstance(data_selector, str):
+            if data_selector not in self.terms:
+                raise KeyError(f"data_selector {data_selector} invalid. Check "
+                               "the EDRReader's `terms` attribute.")
+            data_dict = {"Time": self.auxdata["Time"],
+                         data_selector: self.auxdata[data_selector]}
+            return data_dict
+        else:
+            raise ValueError(f"data_selector of type {type(data_selector)} is "
```

Comment:
Should we be doing this level of type checking? Would we not be ok just returning the KeyError here? Maybe we can just do a try/except around the data_dict keying and catch the KeyError instead?